llvm/lib/Target/AArch64/AArch64Features.td

```diff
@@ -797,6 +794,9 @@ def FeatureUseFixedOverScalableIfEqualCost: SubtargetFeature<"use-fixed-over-sca
   "UseFixedOverScalableIfEqualCost", "true",
   "Prefer fixed width loop vectorization over scalable if the cost-model assigns equal costs">;
 
+def FeatureDontUseScalarIncVL : SubtargetFeature<"dont-use-scalar-inc-vl",
```

Comment:
Can you implement this without changing the feature name?

Reply:
Yeah that should be OK. It should at least make the patch simple and I think just involves inverting all the logic, so turning the option on when we have SVE2/SME (as opposed to turning it off when we have only SVE-1). Having it default to addvl felt like a better default.  Do you have a particular reason to not want to change it?

(I'm not sure if dont-use-scalar-inc-vl is the best name, to be honest. It might be better to have something like preferAddCntToAddvl to avoid the negatives and make it a little more explicit)

Reply:
I found the double negative in `-mattr=-dont-use-scalar-inc-vl` confusing and the current logic is a bit simpler (although I'm not particularly happy with the explicit partial string matching, which I guess is the downside of doing it this way). A name change would avoid the double negative, but thinking about this; why don't we just make it an LLVM option rather than a feature? 

That way, we can do something like:
```
if (ForceScalarIncVL.getNumOccurrences())
  UseScalarIncVL = ForceScalarIncVL.getValue();
else if (hasSVE2() || hasSME())
  UseScalarIncVL = true;
// else UseScalarIncVL = false; (would be default)
```

We can still make use of the `UseScalarIncVL`  Predicate in TableGen, it just wouldn't be defined through a target feature.

Reply:
Yeah the string matching isn't the greatest. We only do it once though, and I believe this is the only way to check if a feature was specified. I assume that it was made a target feature so that we could add it to certain cpus if it is useful. I'm not 100% sure we won't find that would still be useful in some cases, but we have progressed a bit in time now and it hasn't come up yet.